onset_maproom/maproom.py

```diff
@@ -119,10 +215,45 @@ def map_click(click_lat_lng):
     ), round(lat_lng[0],4), round(lat_lng[1],4)
 
 
+@APP.callback(
+    Output("map_title", "children"),
+    Input("search_start_day", "value"),
+    Input("search_start_month", "value"),
+    Input("yearly_stats_input", "value"),
+    Input("probExcThresh1", "value")
+)
+
+def write_map_title(search_start_day, search_start_month, yearly_stats_input, probExcThresh1):
+
+    if yearly_stats_input == "monit":
+        search_start_month1 = calc.strftimeb2int(search_start_month)
+        first_day = rr_mrg.precip["T"][-366:-1].where(
+            lambda x: (x.dt.day == int(search_start_day)) & (x.dt.month == search_start_month1),
+            drop=True,
+        )[0].dt.strftime("%Y-%m-%d").values
```

Comment:
This `where` condition is repeated in lines 661-664. Is it worth factoring out the common code into a function?

Reply:
Not feeling completely confident I implemented this in the most generalizable way... If we wanted to apply it elsewhere...

Reply:
> Not feeling completely confident I implemented this in the most generalizable way.

Maybe not, but it looks good for now.